src/modules/75-cf/components/GitSyncSetupModal/GitSyncSetupModal.tsx

```diff
@@ -0,0 +1,128 @@
+import type { FormikProps } from 'formik'
+import * as Yup from 'yup'
+import { Button, ButtonVariation, Formik, FormikForm, Heading, Layout, ModalDialog, useToaster } from '@harness/uicore'
+import { fullYamlPathRegex } from '@common/utils/StringUtils'
+import { useStrings } from 'framework/strings'
+import { GitSyncForm, GitSyncFormFields } from '@gitsync/components/GitSyncForm/GitSyncForm'
+import { useCreateGitRepo } from 'services/cf'
+import { getErrorMessage } from '@cf/utils/CFUtils'
+
+interface GitSyncSetupModalProps {
+  isOpen: boolean
+  hideModal: () => void
+}
+
+export const GitSyncSetupModal: React.FC<GitSyncSetupModalProps> = ({ isOpen, hideModal }) => {
+  const { orgIdentifier, projectIdentifier, accountId: accountIdentifier } = useParams<Record<string, string>>()
+  const { showError } = useToaster()
+  const { getString } = useStrings()
+
+  const formikRef = useRef<FormikProps<GitSyncFormFields>>()
+
+  const { mutate: createGitRepo } = useCreateGitRepo({
+    identifier: projectIdentifier,
+    queryParams: {
+      accountIdentifier,
+      orgIdentifier
+    }
+  })
+
+  const handleSubmit = useCallback(
+    async (formValues: GitSyncFormFields) => {
+      // get file name and root folder path from filePath field
+      const fullPath = formValues?.filePath
+      const pathNodes = fullPath?.split('/').filter((node: string) => !!node) // remove empty
+      const yamlFileName = '/' + pathNodes?.pop()
+      const yamlFilePath = pathNodes?.length ? '/' + pathNodes?.join('/') + '/' : ''
+
+      const requestData = {
+        repoIdentifier: formValues.repo,
+        repoReference: formValues.repo,
+        branch: formValues.branch,
+        filePath: yamlFileName,
+        rootFolder: yamlFilePath,
+        connectorReference:
+          typeof formValues.connectorRef === 'string' ? formValues.connectorRef : formValues.connectorRef?.value
+      }
+
+      try {
+        await createGitRepo(requestData)
+
+        hideModal()
+      } catch (error) {
+        showError(getErrorMessage(error), 0, getString('cf.selectFlagRepo.createRepoError'))
+      }
+    },
+    [createGitRepo, getString, hideModal, showError]
+  )
+
+  return (
+    <ModalDialog
+      isOpen={isOpen}
+      enforceFocus={false}
+      title={
+        <Heading level={4} font={{ variation: FontVariation.H4 }} color={Color.GREY_800}>
+          {getString('cf.gitSync.setUpGitConnection')}
+        </Heading>
+      }
+      footer={
+        <Layout.Horizontal spacing="small">
+          <Button
+            variation={ButtonVariation.PRIMARY}
+            text={getString('save')}
+            intent="primary"
+            onClick={() => {
+              formikRef?.current?.submitForm()
+            }}
+          />
+          <Button variation={ButtonVariation.TERTIARY} text={getString('cancel')} onClick={hideModal} />
+        </Layout.Horizontal>
+      }
+      onClose={hideModal}
+    >
+      <Formik
+        formName="ffGitExForm"
```

Comment:
I prefer the actual `Formik` and `Form` from the Formik package itself. The UICore version is a bit 💩 IMO.

It's fine as-is, but things are nicer if you use the non UICore version

```suggestion
      <Formik<GitSyncFormFields>
```